vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -204,6 +215,23 @@ type internal FSharpProjectOptionsManager
         | true, (_loadTime, originalOptions) -> Some originalOptions
         | _ -> this.TryGetOptionsForProject(projectId) 
 
+    [<Export>]
+    /// This handles commandline change notifications from the Dotnet Project-system
+    member this.HandleCommandLineChanges(path:string, sources:ImmutableArray<CommandLineSourceFile>, references:ImmutableArray<CommandLineReference>, options:ImmutableArray<string>) =
+        let fullPath p =
+            if Path.IsPathRooted(p) then p
+            else Path.Combine(Path.GetDirectoryName(path), p)
+        let sourcePaths = sources |> Seq.map(fun s -> fullPath s.Path) |> Seq.toArray
+        let referencePaths = references |> Seq.map(fun r -> fullPath r.Reference) |> Seq.toArray
+        projectInfo.[path] <- (sourcePaths,referencePaths,options.ToArray())
+        notify.Trigger(path)
```

Comment:
This all seems contorted, making event spaghetti where it's  not needed. You  have

    NewProjectSystem --> calls ProjectOptionsManager (HandleCommandLineChange) --> notify FSharpLanguageService via event --> calls ProjectOptionsManager (UpdateProjectInfo)

Better would surely be

    NewProjectSystem --> calls FSharpLanguageService (HandleCommandLineChange) --> calls ProjectOptionsManager (UpdateProjectInfo)

With no need for events to implement what is basically a call.